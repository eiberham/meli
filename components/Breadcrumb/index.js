import React from 'react';
import { Breadcrumb as Nav } from 'semantic-ui-react';

const Breadcrumb = props => {
    const { links } = props;
    return (
        <Nav>
            {links && links.map((link, index) => (
                <React.Fragment>
                    {index === links.length -1 ? (
                        <Nav.Section active>{ link }</Nav.Section>
                    ) : (
                        <React.Fragment>
                            <Nav.Section link>{ link }</Nav.Section>
                            <Nav.Divider>/</Nav.Divider>
                        </React.Fragment>
                    )}
                </React.Fragment>
            ))}
        </Nav>
    )
}

export default Breadcrumb;