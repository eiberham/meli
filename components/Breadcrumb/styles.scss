.breadcrumb {
    grid-area: breadcrumb;
    width: 100%;
    margin: 0 auto;
    padding: 1rem 0;
}