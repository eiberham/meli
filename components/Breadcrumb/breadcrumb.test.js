import React from 'react'
import { mount } from 'enzyme'
import Breadcrumb from './index';

describe('Breadcrumb', () => {
    it('renders without crashing', () => {
        const component = mount(<Breadcrumb />);
        expect(component.find('.breadcrumb')).toHaveLength(1);
    })
})