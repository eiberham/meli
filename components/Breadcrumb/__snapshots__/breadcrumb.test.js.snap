// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb matches snapshot 1`] = `
<div
  className="breadcrumb"
>
  <div
    className="ui small breadcrumb"
  >
    <a
      className="section"
      onClick={[Function]}
    >
      Root
    </a>
    <i
      aria-hidden="true"
      className="right angle icon divider"
      onClick={[Function]}
    />
    <a
      className="section"
      onClick={[Function]}
    >
      Item
    </a>
    <i
      aria-hidden="true"
      className="right angle icon divider"
      onClick={[Function]}
    />
    <div
      className="active section"
      onClick={[Function]}
    >
      Subitem
    </div>
  </div>
</div>
`;
